packages/utilities/src/helpers/sort-by.ts

```diff
@@ -89,3 +89,69 @@ export const sortByVersion = (
 
   return order === 'asc' ? result : -result;
 };
+
+/**
+ * Compares two semantic version strings based on the specified order, including with special handling of LKE-Enterprise tier versions.
+ *
+ * This function splits each version string into its constituent parts (major, minor, patch),
+ * compares them numerically, and returns a positive number, zero, or a negative number
+ * based on the specified sorting order. If components are missing in either version,
+ * they are treated as zero.
+ *
+ * @param {string} a - The first version string to compare.
+ * @param {string} b - The second version string to compare.
+ * @param {SortOrder} order - The intended sort direction of the output; 'asc' means lower versions come first, 'desc' means higher versions come first.
+ * @returns {number} Returns a positive number if version `a` is greater than `b` according to the sort order,
+ *                   zero if they are equal, and a negative number if `b` is greater than `a`.
+ * * @example
+ * // returns a positive number
+ * sortByVersion('1.2.3', '1.2.2', 'asc');
+ * sortByVersion('v1.2.3+lke1', 'v1.2.2+lke2', 'asc');
+ *
+ * @example
+ * // returns zero
+ * sortByVersion('1.2.3', '1.2.3', 'asc');
+ * sortByVersion('v1.2.3+lke1', 'v1.2.3+lke1', 'asc');
+ *
+ * @example
+ * // returns a negative number
+ * sortByVersion('1.2.3', '1.2.4', 'asc');
+ * sortByVersion('v1.2.3+lke1', 'v1.2.4+lke1', 'asc');
+ */
+export const sortByTieredVersion = (
```

Comment:
```suggestion
export const compareKubernetesVersion  = (
```

Wondering if we should name this function without the word `sort` because it returns a value for purposes of sorting rather than actually sorting

Reply:
That's fair and a good improvement for developer clarity. It left me with the question of with what to about with the rest of the 'sortBy' functions in `sort-by.ts`, which also do a similar thing, and changing those was too much scope-creep. Ultimately, I made the name change for this new function, moved it to kubeUtils since it's kube specific, and left the rest of the 'sortBy' functions as-is in their original file. ([5702f69](https://github.com/linode/manager/pull/12246/commits/5702f69074bd1b0e54810a3ae7e462e3e3bc0709))